pkg/cluster/cluster.go

```diff
@@ -118,12 +117,16 @@ func (c *Cluster) setup() error {
 		return fmt.Errorf("invalid cluster spec: %v", err)
 	}
 
-	var phase spec.ClusterPhase
+	// Set status here because if the below state check fails, it will use the status field
+	// to also check if it's already "Failed".
```

Comment:
// recover the cached cluster state from the status in the current spec when possible.